src/clj_easy/graal_build_time.clj

```diff
@@ -6,22 +6,33 @@
    :methods [^{:static true} [packageList [java.util.List] "[Ljava.lang.String;"]
              ^{:static true} [packageListStr ["[Ljava.lang.String;"] String]]))
 
-(defn keep-fn [nm]
-  (when (and
-         (not (str/starts-with? nm "clojure"))
-         (str/ends-with? nm "__init.class"))
-    (-> nm
-        (str/replace #"/.*__init.class$" "")
-        (str/replace "/" "."))))
+(defn jar-file-entry->package [nm]
+  (->> (str/split nm (re-pattern (System/getProperty "file.separator")))
+       drop-last
+       (str/join ".")))
+
+(defn ^:private consider-jar-file-entry? [nm]
+  (and (not (str/starts-with? nm (str "clojure" (System/getProperty "file.separator"))))
+       (str/ends-with? nm "__init.class")))
+
+(defn ^:private contains-parent? [packages package]
+  (some #(and (not= % package)
+              (str/starts-with? package %))
+        packages))
 
 (defn packages-from-jar
   [^Path jar-file]
   (with-open [jar (JarFile. (.toFile jar-file))]
     (let [entries (enumeration-seq (.entries jar))
-          packages (keep (fn [^JarFile$JarFileEntry x]
-                           (let [nm (.getName x)]
-                             (keep-fn nm))) entries)]
-      (vec packages))))
+          packages (->> entries
+                        (map #(.getName ^JarFile$JarFileEntry %))
+                        (filter consider-jar-file-entry?)
+                        (map jar-file-entry->package)
+                        (remove str/blank?))
```

Comment:
BTW this is already removing single segment packages like digest

Reply:
Yes, but removing it entirely won't fix the problem for digest, will it?

Reply:
No, I'd go with your first suggestion
> We can require libraries to not use single segment namespaces anymore if they want to use this (graal-build-time) library, which in this day and age is not unreasonable.

Reply:
just not hard requiring in this lib, we will just not include the single segment lib

Reply:
We could maybe warn something for single-segment libs?

Reply:
Yes, or at least document it and that they won't be supported: just change your library to good standards. I will also implement a linter for this in clj-kondo.

Reply:
Let me know if this is good for merge btw.